src/Rpc.re

```diff
@@ -158,8 +161,10 @@ let start =
             let str = Bytes.to_string(buffer);
             let result = _parse(str);
 
-            scheduler(() => rpc.messageHandler(result, rpc));
-            ();
+            scheduler(() => {
+              Log.debugf(m => m("Received: %s", str));
```

Comment:
I also made a small change here - I moved the `Log.debugf` inside the `scheduler` call. 

There might be a better approach (ie, perhaps the Log methods should be thread-safe?) but in Onivim 2, we use `Revery.App.runOnMainThread` as the scheduler. If we don't 'schedule' this call, we could run into those race conditions where the logging gets interleaved & the color gets corrupted.